tests/envs/test_register.py

```diff
@@ -0,0 +1,182 @@
+    with pytest.raises(gym.error.Error, match="Malformed environment ID"):
+        gym.register(env_id)
+
+
+@pytest.mark.parametrize(
+    "env_id_input, env_id_suggested",
+    [
+        ("cartpole-v1", "CartPole"),
+        ("blackjack-v1", "Blackjack"),
+        ("Blackjock-v1", "Blackjack"),
+        ("mountaincarcontinuous-v0", "MountainCarContinuous"),
+        ("taxi-v3", "Taxi"),
+        ("taxi-v30", "Taxi"),
+        ("MyAwesomeNamspce/MyAwesomeVersionedEnv-v1", "MyAwesomeNamespace"),
+        ("MyAwesomeNamspce/MyAwesomeUnversionedEnv", "MyAwesomeNamespace"),
+        ("MyAwesomeNamespace/MyAwesomeUnversioneEnv", "MyAwesomeUnversionedEnv"),
+        ("MyAwesomeNamespace/MyAwesomeVersioneEnv", "MyAwesomeVersionedEnv"),
+    ],
+)
+def test_env_suggestions(register_testing_envs, env_id_input, env_id_suggested):
+    with pytest.raises(
+        gym.error.UnregisteredEnv, match=f"Did you mean: `{env_id_suggested}`?"
+    ):
+        gym.make(env_id_input)
+
+
+@pytest.mark.parametrize(
+    "env_id_input, suggested_versions, default_version",
+    [
+        ("CartPole-v12", "`v0`, `v1`", False),
+        ("Blackjack-v10", "`v1`", False),
+        ("MountainCarContinuous-v100", "`v0`", False),
+        ("Taxi-v30", "`v3`", False),
+        ("MyAwesomeNamespace/MyAwesomeVersionedEnv-v6", "`v1`, `v3`, `v5`", False),
+        ("MyAwesomeNamespace/MyAwesomeUnversionedEnv-v6", "", True),
+    ],
+)
+def test_env_version_suggestions(
+    register_testing_envs, env_id_input, suggested_versions, default_version
+):
+    if default_version:
+        match_str = "provides the default version"
+        with pytest.raises(
+            gym.error.DeprecatedEnv,
+            match=match_str,
+        ):
+            gym.make(env_id_input)
+    else:
+        match_str = f"versioned environments: \\[ {suggested_versions} \\]"
+        with pytest.raises(gym.error.UnregisteredEnv, match=match_str):
+            gym.make(env_id_input)
+
+
+def test_register_versioned_unversioned():
+    # Register versioned then unversioned
+    versioned_env = "Test/MyEnv-v0"
+    gym.register(versioned_env)
+    assert gym.envs.spec(versioned_env).id == versioned_env
+
+    unversioned_env = "Test/MyEnv"
+    with pytest.raises(
+        gym.error.RegistrationError, match="Can't register the unversioned environment"
+    ):
+        gym.register(unversioned_env)
+
+    # Clean everything
+    del gym.envs.registry[versioned_env]
+
+    # Register unversioned then versioned
+    gym.register(unversioned_env)
+    assert gym.envs.spec(unversioned_env).id == unversioned_env
+    with pytest.raises(gym.error.RegistrationError):
+        gym.register(versioned_env)
+
+    # Clean everything
+    del gym.envs.registry[unversioned_env]
+
+
+def test_make_latest_versioned_env(register_testing_envs):
+    with pytest.warns(UserWarning, match="Using the latest versioned environment"):
+        env = gym.make("MyAwesomeNamespace/MyAwesomeVersionedEnv")
```

Comment:
The original code had `disable_env_checker=True`. Why did that go?

Reply:
Fixed